src/fides/api/alembic/migrations/helpers/fideslang_migration_functions.py

```diff
@@ -0,0 +1,451 @@
+            )
+            bind.execute(
+                update_query,
+                {"notice_id": row["id"], "updated_uses": updated_data_uses},
+            )
+
+
+def update_consent(bind: Connection, data_use_map: Dict[str, str]) -> None:
+    """
+    Update Consent objects in the database.
+    """
+
+    # Update the Consent table
+    existing_consents: ResultProxy = bind.execute(
+        text("SELECT provided_identity_id, data_use FROM consent;")
+    )
+
+    for row in existing_consents:
+        updated_use: str = _replace_matching_data_label(row["data_use"], data_use_map)
+
+        update_label_query: TextClause = text(
+            "UPDATE consent SET data_use= :updated_label WHERE provided_identity_id= :key AND data_use = :old_use"
+        )
+        bind.execute(
+            update_label_query,
+            {
+                "key": row["provided_identity_id"],
+                "old_use": row["data_use"],
+                "updated_label": updated_use,
+            },
+        )
+
+    # Update the Privacy Request Table
+    existing_privacy_requests: ResultProxy = bind.execute(
+        text("select id, consent_preferences from privacyrequest;")
+    )
+
+    for row in existing_privacy_requests:
+        preferences: List[Dict] = row["consent_preferences"]
+
+        if preferences:
+            for index, preference in enumerate(preferences):
+                preferences[index]["data_use"] = _replace_matching_data_label(
+                    data_label=preference["data_use"], data_label_map=data_use_map
+                )
+
+        update_pr_query: TextClause = text(
+            "UPDATE privacyrequest SET consent_preferences= :updated_preferences WHERE id= :id"
+        )
+        bind.execute(
+            update_pr_query,
+            {"id": row["id"], "updated_preferences": json.dumps(preferences)},
+        )
+
+    # Update the Consent Request Table
+    existing_consent_requests: ResultProxy = bind.execute(
+        text("select id, preferences from consentrequest;")
+    )
+
+    for row in existing_consent_requests:
+        preferences: List[Dict] = row["preferences"]
+
+        if preferences:
+            for index, preference in enumerate(preferences):
+                preferences[index]["data_use"] = _replace_matching_data_label(
+                    data_label=preference["data_use"], data_label_map=data_use_map
+                )
+
+        update_cr_query: TextClause = text(
+            "UPDATE consentrequest SET preferences= :updated_preferences WHERE id= :id"
+        )
+        bind.execute(
+            update_cr_query,
+            {"id": row["id"], "updated_preferences": json.dumps(preferences)},
+        )
+
+
+def remove_conflicting_rule_targets(db: Session):
+    """
+    Iterates through all of the erasure policies and removes level 3 data categories in favor of level 2 data categories.
+
+    For example: user.demographic is preserved over user.demographic.*
```

Comment:
It took me awhile to track down why this was needed (it's been awhile), a code comment might be helpful.